app/apps.py

```diff
@@ -110,17 +110,18 @@ def search(query: str):
     if results := db.search(query):
         appids = tuple(doc_id.replace("fts", "apps") for doc_id in results)
         apps = [json.loads(x) for x in db.redis_conn.mget(appids)]
-
         ret = []
+        downloadid = []
```

Comment:
This is probably not what you wanted to do. You could just drop this and define the variable (not a list) inside the for scope.

Reply:
I used a list because the function`get_total_downloads_by_ids` requires a list to work. Would changing that function be a better alternative?

Reply:
Well, you've added that function, do what you like.

You could also keep it like that and just do `stats.get_total_downloads_by_ids([app["id"]])`.

The thing is your current implementation will do a bunch of unneeded work, as your list will, grow and you will fetch downloads again and again.

- "id1000" -> get downloads
- "id1000", "id1001" -> get downloads
- "id1000", "id1001", "id1002" -> get downloads
- etc